sphericart-jax/src/jax.cpp

```diff
@@ -0,0 +1,90 @@
+// This file defines the Python interface to the XLA custom call implemented on the CPU.
+// It is exposed as a standard pybind11 module defining "capsule" objects containing our
+// method. For simplicity, we export a separate capsule for each supported dtype.
+
+#include <cstdlib>
+#include <iostream>
+
+#include "sphericart.hpp"
+#include "sphericart/pybind11_kernel_helpers.h"
+
+using namespace sphericart_jax;
+
+namespace {
+
+template <typename T>
+void cpu_sph(void *out, const void **in) {
+    // std::cout << "Called kernel" << std::endl;
+    // Parse the inputs
+    const T *xyz = reinterpret_cast<const T*>(in[0]);
+    const size_t l_max = *reinterpret_cast<const int*>(in[1]);
+    const bool normalized = *reinterpret_cast<const bool*>(in[2]);
+    const size_t n_samples = *reinterpret_cast<const int*>(in[3]);
+    // std::cout << l_max << " " << normalized << " " << n_samples << std::endl;
+    size_t xyz_length{n_samples*3};
+    size_t sph_len{(l_max+1)*(l_max+1)*n_samples};
+
+    // The output is stored as a single pointer since there is only one output,
+    // hence no need to reinterpret_cast the out pointer
+    T *sph = reinterpret_cast<T *>(out);  // CHANGEEEEEEEEEEEEEEE SHOULD NOT NEED REINTERPRET_CAST
```

Comment:
So what do we do here?

Reply:
I was wrong: it does need a cast from void* to T*